pkgs/desktops/lxqt/lxqt-wayland-session/default.nix

```diff
@@ -0,0 +1,61 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  kwindowsystem,
+  liblxqt,
+  libqtxdg,
+  lxqt-build-tools,
+  pkg-config,
+  qtsvg,
+  qttools,
+  wrapQtAppsHook,
+  xdg-user-dirs,
+  gitUpdater,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lxqt-wayland-session";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "lxqt";
+    repo = "lxqt-wayland-session";
+    rev = version;
+    hash = "sha256-5WdfwJ89HWlXL6y9Lpgs7H3mbN/wbf+9VbP9ERPasBM=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    lxqt-build-tools
+    pkg-config
+    qttools
+    wrapQtAppsHook
```

Comment:
lxqt-wayland-session seems to be just a collection of configurations. Does it include any Qt programs?

Reply:
``` console
$ find /nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/bin
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/bin/startlxqtwayland
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/bin/.startlxqtwayland-wrapped
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/nix-support
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/nix-support/propagated-user-env-packages
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/desk.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/menu.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/themerc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/iconify.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/shade.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/desk_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/max_toggled_hover.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/max.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/close.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/max_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/shade_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/desk.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/menu.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/themerc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/iconify.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/shade.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/desk_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/max_toggled_hover.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/max.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/close.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/max_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/shade_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wallpapers
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wallpapers/origami-dark-labwc.png
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/themerc-override
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/rc.xml
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/autostart
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/menu.xml
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/themerc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/environment
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/README
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/firstrun
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/firstrun/autostart
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-niri.kdl
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-river-init
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-hyprland.conf
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-wayfire.ini
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-sway.config
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/graphics
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/graphics/lxqt-labwc.png
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/wayland-sessions
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/wayland-sessions/lxqt-wayland.desktop
```

`bin/.startlxqtwayland-wrapped` is a shell script

Reply:
So no Qt program is included.

Reply:
I have now removed the dependence on `wrapQtAppsHook`.

Reply:
There are some hardcoded paths in this project that you need to pay attention to, eg：

https://github.com/lxqt/lxqt-wayland-session/blob/c8cbc996b58bce24c2564b62b99c4f6970106bf3/startlxqtwayland.in#L82

https://github.com/lxqt/lxqt-wayland-session/blob/c8cbc996b58bce24c2564b62b99c4f6970106bf3/configurations/labwc/autostart#L6

Reply:
Which is more appropriate: replace `/usr/share/X11/xkb/rules/base.lst` by `${xkeyboard-config}/share/X11/xkb/rules/base.lst` or by `/run/current-system/sw/share/X11/xkb/rules/base.lst`?

Reply:
```
>>> rg /usr/share/X11/xkb/rules/
pkgs/os-specific/linux/systemd/0008-localectl-use-etc-X11-xkb-for-list-x11.patch
20:-        f = fopen("/usr/share/X11/xkb/rules/base.lst", "re");

pkgs/desktops/lxqt/lxqt-config/default.nix
72:      --replace-fail '/usr/share/X11/xkb/rules/base.lst' \

pkgs/desktops/deepin/core/dde-session-shell/default.nix
49:      --replace "/usr/share/X11/xkb/rules/base.xml" "${xkeyboard_config}/share/X11/xkb/rules/base.xml"

pkgs/applications/misc/synergy/default.nix
60:      --replace "/usr/share/X11/xkb/rules/evdev.xml" "${xkeyboardconfig}/share/X11/xkb/rules/evdev.xml"

pkgs/applications/display-managers/sddm/unwrapped.nix
28:      --replace "/usr/share/X11/xkb/rules/evdev.xml" "${xkeyboardconfig}/share/X11/xkb/rules/evdev.xml"

pkgs/tools/misc/calamares/default.nix
72:    sed -e 's,/usr/share/X11/xkb/rules/base.lst,${xkeyboard_config}/share/X11/xkb/rules/base.lst,' \

```

`${xkeyboard_config}/share/X11/xkb/rules/base.lst ` should be fine